File: jvm/core/main/scala/com/webank/eggroll/core/containers/container/PythonContainer.scala
/*
 * Copyright (c) 2019 - now, Eggroll Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 *
 */

package com.webank.eggroll.core.containers.container

import java.nio.file.Path

class PythonContainer(
                       pythonExec: String,
                       scriptPath: String,
                       cwd: Path,
                       scriptArgs: Seq[String] = Seq.empty,
                       extraEnv: Map[String, String] = Map.empty,
                       stdErrFile: Option[Path] = None,
                       stdOutFile: Option[Path] = None,
                       workingDirectoryPreparer: Option[WorkingDirectoryPreparer] = None,
                       containerId: String,
                       processorId: Long
                     )

  extends ProcessContainer(
    command = Seq(pythonExec, "-u", scriptPath) ++ scriptArgs,
    extraEnv = extraEnv,
    stdOutFile = stdOutFile,
    stdErrFile = stdErrFile,
    cwd = cwd,
    workingDirectoryPreparer = workingDirectoryPreparer,
    containerId = containerId,
    processorId =  processorId
  )

File: jvm/core/src/main/java/org/fedai/eggroll/core/containers/container/PythonContainer.java
package org.fedai.eggroll.core.containers.container;


import java.nio.file.Path;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class PythonContainer extends ProcessContainer {
    private String pythonExec;
    private String scriptPath;
    private Path cwd;
    private List<String> scriptArgs;
    private Map<String, String> extraEnv;
    private Path stdErrFile;
    private Path stdOutFile;
    private WorkingDirectoryPreparer workingDirectoryPreparer;
    private long processorId;

    public PythonContainer(String pythonExec, String scriptPath, Path cwd, List<String> scriptArgs,
                           Map<String, String> extraEnv, Path stdErrFile, Path stdOutFile,
                           WorkingDirectoryPreparer workingDirectoryPreparer, long processorId) {
        super(buildCommand(pythonExec, scriptPath, scriptArgs), cwd, extraEnv, stdOutFile, stdErrFile,
                workingDirectoryPreparer, processorId);

        this.pythonExec = pythonExec;
        this.scriptPath = scriptPath;
        this.cwd = cwd;
        this.scriptArgs = scriptArgs;
        this.extraEnv = extraEnv;
        this.stdErrFile = stdErrFile;
        this.stdOutFile = stdOutFile;
        this.workingDirectoryPreparer = workingDirectoryPreparer;
        this.processorId = processorId;
    }

    private static List<String> buildCommand(String pythonExec, String scriptPath, List<String> scriptArgs) {
        List<String> command = new ArrayList<>();
        command.add(pythonExec);
        command.add("-u");
        command.add(scriptPath);
        command.addAll(scriptArgs);
        return command;
    }
}